tool/teleport/testenv/test_server.go

```diff
@@ -330,6 +330,36 @@ func WithAuthPreference(authPref types.AuthPreference) TestServerOptFunc {
 	})
 }
 
+func SetupTrustedCluster(ctx context.Context, t *testing.T, rootServer, leafServer *service.TeleportProcess, additionalRoleMappings ...types.RoleMapping) {
```

Comment:
Are there other trusted cluster tests that should use this in the future?

I'm concerned about leaving multiple ways to do the same thing in the codebase and having people reintroduce flaky tests in the future by picking the wrong one.

I'm not suggesting any larger changes in this PR, but maybe a comment or TODO would help.

Reply:
I don't think this particular change contributes to the flakiness here, but agreed. I added deprecation comments. Ideally we would only use `tool/teleport/testenv` for test servers so that any fixes apply to all tests.